File: voc-mesh-platform/apps/api/app/models/parcela.py
import uuid
from datetime import datetime, timezone

from sqlalchemy import DateTime, Float, ForeignKey, String
from sqlalchemy.dialects.postgresql import JSON, UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Parcela(Base):
    __tablename__ = "parcelas"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    tenant_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("tenants.id"), nullable=False, index=True
    )
    name: Mapped[str] = mapped_column(String(256), nullable=False)
    crop_type: Mapped[str] = mapped_column(String(128), nullable=False)
    area_km2: Mapped[float] = mapped_column(Float, default=0.0)
    geojson: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=lambda: datetime.now(timezone.utc),
    )

File: voc-mesh-platform/packages/voc-schema/src/parcela.ts
import { z } from "zod";

export const ParcelaSchema = z.object({
  id: z.string().uuid(),
  tenant_id: z.string().uuid(),
  name: z.string(),
  crop_type: z.string(),
  area_km2: z.number(),
  geojson: z.record(z.any()),
});
export type Parcela = z.infer<typeof ParcelaSchema>;
